src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerIndexServiceTest.kt

```diff
@@ -297,6 +300,148 @@ class PrisonerIndexServiceTest {
+      prisonerIndexService.compareAndMaybeIndexPrisoner(prisonerId = "A1234AA")
+
+      verify(incentivesService).getCurrentIncentive(123456L)
+    }
+
+    @Test
+    internal fun `will not get incentive if there is no booking`() {
+      whenever(nomisService.getOffender("A1234AA")).thenReturn(anOffenderWithNoBooking())
+      prisonerIndexService.compareAndMaybeIndexPrisoner(prisonerId = "A1234AA")
+
+      verifyNoInteractions(incentivesService)
+    }
+
+    @Test
+    internal fun `will do nothing if no differences found`() {
+      whenever(nomisService.getOffender("A1234AA")).thenReturn(anOffenderBooking(123456L))
+      whenever(prisonerDifferenceService.prisonerHasChanged(any(), any())).thenReturn(false)
+      whenever(prisonerARepository.findById(any())).thenReturn(Optional.of(PrisonerA()))
+
+      prisonerIndexService.compareAndMaybeIndexPrisoner(prisonerId = "A1234AA")
+
+      verify(prisonerARepository, never()).save(any())
+      verify(prisonerDifferenceService).prisonerHasChanged(any(), any())
+      verifyNoMoreInteractions(prisonerDifferenceService)
+    }
+
+    @Test
+    internal fun `will report differences if found`() {
+      whenever(nomisService.getOffender("A1234AA")).thenReturn(anOffenderBooking(123456L))
+      whenever(prisonerDifferenceService.prisonerHasChanged(any(), any())).thenReturn(true)
+      whenever(prisonerARepository.findById(any())).thenReturn(Optional.of(PrisonerA()))
+
+      prisonerIndexService.compareAndMaybeIndexPrisoner(prisonerId = "A1234AA")
+
+      verify(prisonerDifferenceService).reportDiffTelemetry(any(), any())
+      verify(prisonerDifferenceService).handleDifferences(any(), any(), any())
+    }
+
+    @Test
+    internal fun `will update index with current incentive level`() {
+      whenever(nomisService.getOffender("A1234AA")).thenReturn(anOffenderBooking(123456L))
+      whenever(incentivesService.getCurrentIncentive(123456L)).thenReturn(
+        IncentiveLevel(
+          iepCode = "STD",
+          iepLevel = "Standard",
+          iepTime = LocalDateTime.parse("2021-01-01T10:00:00.169539"),
+          nextReviewDate = LocalDate.parse("2021-03-01"),
+        ),
+      )
+      whenever(prisonerARepository.findById(any())).thenReturn(Optional.of(PrisonerA()))
+      whenever(prisonerDifferenceService.prisonerHasChanged(any(), any())).thenReturn(true)
+
+      prisonerIndexService.compareAndMaybeIndexPrisoner(prisonerId = "A1234AA")
+      verify(prisonerARepository).save(
+        check {
+          assertThat(it.currentIncentive?.level?.code).isEqualTo("STD")
+          assertThat(it.currentIncentive?.level?.description).isEqualTo("Standard")
+          assertThat(it.currentIncentive?.dateTime).isEqualTo(LocalDateTime.parse("2021-01-01T10:00:00"))
+          assertThat(it.currentIncentive?.nextReviewDate).isEqualTo(LocalDate.parse("2021-03-01"))
+        },
+      )
+    }
+
+    @Test
+    internal fun `will get restrictive patients data is prisoner OUT`() {
```

Comment:
That old `restrictive` typo

Reply:
fixed some of them, but still classes etc. with typo